src/components/DateInput.jsx

```diff
@@ -81,13 +87,29 @@ export default class DateInput extends React.Component {
   }
 
   onKeyDown(e) {
```

Comment:
it might be valuable to debounce this event so that it doesn't get triggered as frequently when the user is pressing down a key, especially if they're doing a key combination